app/services/rhapsody_realities.js

```diff
@@ -89,10 +89,53 @@ class RhapsodyService {
                 };
             })
             .catch(err => {
-                this.logger.error('failed to fetch Rhasodies ', err.message);
+                this.logger.error('failed to fetch Rhapsodies ', err.message);
                 throw new errors.UnknownError('an unknown error occurred');
             });
     };
+
+    /**
+     * Get one rhapsody by id
+     */
+    getRhapsody(rhapsodyId) {
+        return new Rhapsody({ id: rhapsodyId }).fetch()
```

Comment:
we need to trigger an error if the rhapsody we are trying to fetch is not found in the db

you have 2 ways to deal with this, 
1. hijack the fetch in the model to return a custom error e.g
```fetch: function() {
        return Bookshelf.Model.prototype.fetch.apply(this, arguments)
        .catch(error => {
            if (error instanceof Rhapsody.NotFoundError) {
                throw new errors.RhapsodyNotFound('Rhapsody not found');
            }

            throw error;
        });
    }```

2. pass an argument to fetch like so `.fetch({required: false})` see more info [here](http://bookshelfjs.org/#Model-instance-fetch)

no. 1 is preferred (check the User model in the project i shared with you to see this in action ) , because it follows the single responsibility principle ( since we define it in a single place i.e the model and its automatically used across everywhere we do fetching )

Reply:
done